nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/c2/C2NifiClientService.java

```diff
@@ -90,15 +96,18 @@ public C2NifiClientService(final NiFiProperties niFiProperties, final FlowContro
         this.heartbeatPeriod = clientConfig.getHeartbeatPeriod();
         this.flowController = flowController;
         C2HttpClient client = new C2HttpClient(clientConfig, new C2JacksonSerializer());
-        C2HeartbeatFactory heartbeatFactory = new C2HeartbeatFactory(clientConfig, flowIdHolder);
+        C2HeartbeatFactory heartbeatFactory = new C2HeartbeatFactory(clientConfig, flowIdHolder, new ManifestHashProvider());
+        OperandPropertiesProvider emptyOperandPropertiesProvider = new EmptyOperandPropertiesProvider();
```

Comment:
A question: do we need to inject operandPropertiesProvider from here to keep the C2 layer decoupled from Minifi?
My feeling is that this class start to get a bit crowded, and as a glue between Minifi and C2 I expect it to have even more logic in the future. I think we should think about how we could extract logic into separate classes (not strictly in the PR)

Reply:
Yes, for now, it doesn't have too much added value because neither of the supported operations has any additional properties, but in the near future, I'm planning to add the update properties command where I think it'll be better to pass these to the C2 client.